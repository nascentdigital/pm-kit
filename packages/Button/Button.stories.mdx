import { Meta, Story, Preview, Props, Subtitle } from '@storybook/addon-docs/blocks'
import Button from './Button.jsx'
import { action } from '@storybook/addon-actions'
import { version } from './package.json'

<Meta title="Design System|Button" componenent={Button}/>

# Button
<Subtitle>
version {version}
</Subtitle>


## Usage
* To install `Button` package with npm run terminal:

```jsx
npm i @pm-kit/button
```
* Import `Button` in your componenent to render

```jsx
import Button from '@pm-kit/button'
```

## Properties 

<Props of={Button} />

## Stories

Primary
<Preview>
<Story name="Primary">
  <Button onClick={action('clicked')}>Primary Button</Button>
</Story>
</Preview>

Secondary
<Preview>
<Story name="Secondary">
  <Button variant="secondary" onClick={action('clicked')}>
    Secondary Button
  </Button>
</Story>
</Preview>

Inverted
<Preview>
<Story name="Inverted">
  <Button variant="inverted" onClick={action('clicked')}>
    Inverted Button
  </Button>
</Story>
</Preview>

Disabled
<Preview>
<Story name="Disabled">
  <Button disabled onClick={action('clicked')}>
    Disabled Button
  </Button>
</Story>
</Preview>

Alternative
<Preview>
<Story name="Alternative">
  <Button variant="alternative" onClick={action('clicked')}>
    Alternative Button
  </Button>
</Story>
</Preview>