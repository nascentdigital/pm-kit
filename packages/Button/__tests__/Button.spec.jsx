import React from 'react'
import { shallow } from 'enzyme'

import Button from '../Button'

describe('Button', () => {
  const doShallow = (overrides = {}) => {
    const button = shallow(<Button {...overrides}>Submit</Button>)

    return button
  }

  afterEach(() => {
    jest.clearAllMocks()
  })

  it('renders', () => {
    const button = shallow(<Button>Submit</Button>)

    expect(button).toMatchSnapshot()
  })

  it('has one of the HTML button types', () => {
    let button = doShallow()
    expect(button).toHaveProp('type', 'button')

    button = doShallow({ type: 'reset' })
    expect(button).toHaveProp('type', 'reset')
  })
})
