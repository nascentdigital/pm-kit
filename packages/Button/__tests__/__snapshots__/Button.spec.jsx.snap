// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders 1`] = `
<Button
  forwardedRef={null}
  type="button"
  variant="primary"
  wide={false}
>
  Submit
</Button>
`;
